lib/OpenQA/WebAPI/Controller/API/V1/Search.pm

```diff
@@ -149,10 +149,23 @@ sub _search_job_templates {
     my @results;
     my $last_group = -1;
     my $like       = {like => "%${keywords}%"};
-    my $templates  = $self->schema->resultset('JobTemplates')->search({-or => {name => $like, description => $like}})
-      ->slice(0, $limit);
+
+    # Take into account names of test suites in cases where the job template itself has no name or description,
+    # but is based off of a test suite
+    my $test_suite_ids = $self->_get_test_suite_ids($keywords);
+    my $templates
+      = $self->schema->resultset('JobTemplates')
+      ->search({-or => {name => $like, description => $like, test_suite_id => {-in => $test_suite_ids}}},
+        {order_by => {-desc => 'group_id'}})->slice(0, $limit);
     while (my $template = $templates->next) {
-        my $contents = $template->name . "\n" . $template->description;
+        my $contents;
+        my $description = $template->description;
+        if (my $template_name = $template->name) {
+            $contents = $template_name . "\n" . $description;
+        }
+        else {
+            $contents = $template->test_suite->name . "\n" . $description;
+        }
```

Comment:
Strange. How could you add this line when it's not covered by tests?

Reply:
Oh, in the *Files Changed* view there's a message `Added line #L167 was not covered by tests`. @okurz Are you asking if this case should be, but isn't covered by a test?

Reply:
added a test to cover this line

Reply:
@cdywan well, that it should be covered by tests was obvious to me. I wonder about how to come into the situation that this isn't covered. Like what was the observation that Amrysliu made or what problem she tried to fix that wasn't shown in tests

Reply:
The message is gone now. I still don't know what the actual question is or was, but it seems @Amrysliu extended the coverage now.